spec/currency_spec.rb

```diff
@@ -186,6 +188,24 @@ def to_s
       it "raises UnknownCurrency with unknown currency" do
         expect { Currency.new("xxx") }.to raise_error(Currency::UnknownCurrency, /xxx/)
       end
+
+      it 'returns old object for the same :key' do
+        expect(Currency.new("USD")).to be(Currency.new("USD"))
+        expect(Currency.new("USD")).to be(Currency.new(:usd))
+        expect(Currency.new("USD")).to be(Currency.new(:USD))
+        expect(Currency.new("USD")).to be(Currency.new('usd'))
+        expect(Currency.new("USD")).to be(Currency.new('Usd'))
+
+        ids = []
+        [ 2.times.map{ Thread.new{ ids << Currency.new("USD").object_id } } ].each(&:join)
```

Comment:
Usd is created 2 lines before. Maybe it'll be better to extract it to separate example.